src/Kiota.Builder/CodeDOM/CodeMethod.cs

```diff
@@ -150,19 +150,26 @@ public PagingInformation PagingInformation
     /// </summary>
     public IEnumerable<CodeParameter> PathQueryAndHeaderParameters
     {
-        get; private set;
+        get => pathQueryAndHeaderParameters.Values;
     }
+    private readonly Dictionary<string, CodeParameter> pathQueryAndHeaderParameters = new(StringComparer.OrdinalIgnoreCase);
     public void AddPathQueryOrHeaderParameter(params CodeParameter[] parameters)
     {
         if (parameters == null || !parameters.Any()) return;
-        foreach (var parameter in parameters)
+        foreach (var parameter in parameters.OrderByDescending(static x => x.Kind)) //guarantees that path parameters are added first and other are deduplicated
         {
             EnsureElementsAreChildren(parameter);
+            if (!pathQueryAndHeaderParameters.TryAdd(parameter.Name, parameter)) {
+                if(parameter.IsOfKind(CodeParameterKind.QueryParameter))
+                    parameter.Name += "-query-parameter";
```

Comment:
I'm wondering if the query parameter would be really long to type? I'm suggesting we change to `-query` instead i.e. `--id-query` instead of `--id-query-parameter`. What do you think?

Reply:
Some parameter names would also be longer than id which could add to the typing fatigue. Ultimately, when we have shell completion support, it won't be as bad.

Reply:
```suggestion
                    parameter.Name += "-query";
```